packit_service/worker/parser.py

```diff
@@ -383,6 +335,11 @@ def parse_push_event(event) -> Optional[PushGitHubEvent]:
 
         if not (raw_ref and head_commit and before and pusher):
             return None
+        elif event.get("deleted"):
```

Comment:
Why not do this even before starting any processing of the event in [`api/webooks`](https://github.com/packit/packit-service/blob/33af9662ba6bb6df94a3fe2a4805b9c746b3e0fa/packit_service/service/api/webhooks.py#L82)? I'm aware that this might be outside the scope of this PR.

Reply:
> I'm aware that this might be outside the scope of this PR.
 
Agree. This is relevant to the filtering of other events as well.

Reply:
I have looked into it, we basically just do validation there, so I didn't want to include this kind of a logic that is specific to push event. It should be discussed further imo.

Reply:
IIRC we already had several discussions on this topic on arch meetings.

The conclusion there was that it's okay to do some filtering on the API level, if that filtering can be done solely based on the information available in the event payload or HTTP headers. This would not require any considerable extra time spent in the API as inspecting a few values in dictionaries is cheap and there wouldn't be any external resources involved (no additional API calls, lengthy disk reads or such).

Yes, an approach like this would cause having two places where events can be filtered out, but it would also save us quite a few extra CPU cycles we currently use just to encode and then decode the whole payload. Furthermore, workers would need to deal only with events where there is actually some extra work to be done (events which are actionable), like fetching a `.packit.yaml` at the minimum.

And I'm pretty sure such an approach would also help with preventing mistakes like this PR tries to fix :slightly_smiling_face:

Reply:
Yes, I recall we want to solve this in general. Not only for this specifically. (But when we discussed this last time, we mentioned that this will require changes in a lot of test cases.)

I would merge this and create an issue/card for the move of filtering to the API, what do you think?

Reply:
Sure :+1: